torch/csrc/jit/codegen/cuda/executor.h

```diff
@@ -310,7 +310,9 @@ class TORCH_CUDA_CU_API FusionExecutor : public NonCopyable {
   // Profiling support: the last launch param used
   LaunchParams launch_params_;
 
-  // Profiling support: knob to disable caching of launch params
+  // Profiling support: disable caching of launch params and output allocation
+  // output allocation is also disable when output sizes are dependent on runtime inputs.
```

Comment:
nit: 
... when output sizes are dependent on runtime scalar inputs, such as for the case of tensor factory.